src/main/java/com/flipkart/zjsonpatch/JsonDiff.java

```diff
@@ -452,10 +441,6 @@ private static void compareObjects(List<Diff> diffs, List<Object> path, JsonNode
     }
 
     private static List<JsonNode> getLCS(final JsonNode first, final JsonNode second) {
-
-        Preconditions.checkArgument(first.isArray(), "LCS can only work on JSON arrays");
-        Preconditions.checkArgument(second.isArray(), "LCS can only work on JSON arrays");
-
-        return ListUtils.longestCommonSubsequence(Lists.newArrayList(first), Lists.newArrayList(second));
+        return InternalUtils.longestCommonSubsequence(InternalUtils.toList((ArrayNode) first), InternalUtils.toList((ArrayNode) second));
```

Comment:
Slight change in error semantics, parameters no longer checked for whether or not they are arrays. Intentional?

Reply:
Since the caller already checks whether both inputs( source & target) are of Array type before calling this method, i thought to remove the additional checks.

Reply:
Makes sense, but I checked and couldn't see the aforementioned check at the caller. Anyway it's only for clarity of errors when the API is misused, so not a critical point regardless.

Reply:
Actually, caller of compareArray() methods checks the data-type of source & target.